src/components/GroupModal/GroupModalReservationsList.tsx

```diff
@@ -75,45 +76,57 @@ const GroupModalReservationsList: FC<GroupModalReservationsListProps> = ({
             faust,
             identifier,
             numberInQueue,
-            reservationIds
+            pickupDeadline,
+            reservationIds,
+            state
           } = material;
           const selected = selectedMaterials?.some((selectedMaterial) =>
             isEqual(selectedMaterial, material)
           );
+
+          let statusText = "";
+
+          if (identifier && state === "reserved") {
+            if (pickupDeadline === null || pickupDeadline === "") {
+              statusText = t("reservationListYouAreInQueueText");
+            } else {
+              statusText = t("reservationListAvailableInText", {
+                placeholders: {
+                  "@count": daysBetweenTodayAndDate(pickupDeadline ?? "")
+                }
+              });
+            }
+          } else if (faust && numberInQueue) {
+            statusText = t("dashboardNumberInLineText", {
+              count: numberInQueue,
+              placeholders: { "@count": numberInQueue }
+            });
+          }
+
+          const statusBadgeComponent = statusText ? (
+            <StatusBadge
+              badgeDate={expiryDate}
+              neutralText={statusText}
+              infoText=""
+            />
+          ) : null;
```

Comment:
I think this should be put into a helper function in a `helper.ts` file inside the `GroupModal` folder.
Personally I am not a big fan of `let` but thats maybe more a matter of taste.
Here is a suggestion for a typed helper function:
```typescript 
export const getStatusText = ({
  identifier,
  state,
  pickupDeadline,
  faust,
  numberInQueue,
  t
}: ReservationType & { t: UseTextFunction }) => {
  if (identifier && state === "reserved") {
    if (pickupDeadline === null || pickupDeadline === "") {
      return t("reservationListYouAreInQueueText");
    }

    return t("reservationListAvailableInText", {
      placeholders: {
        "@count": daysBetweenTodayAndDate(pickupDeadline ?? "")
      }
    });
  }

  if (faust && numberInQueue) {
    return t("dashboardNumberInLineText", {
      count: numberInQueue,
      placeholders: { "@count": numberInQueue }
    });
  }

  return "";
};
```